File: django/app/api/models/detalleProducto.py
from django.db import models

class DetalleProducto(models.Model):

    producto = models.ForeignKey('Producto', 
                                on_delete=models.CASCADE,
                                related_name="producto")
    precio_costo = models.DecimalField(max_digits=8, decimal_places=2)
    precio_venta = models.DecimalField(max_digits=8, decimal_places=2)
    existencias = models.IntegerField(default=0)
    vencimiento = models.CharField(max_length=50, blank=True)
    almacen = models.ForeignKey('Ubicacion', 
                                on_delete=models.CASCADE,
                                related_name="ubi")
    estanteria = models.ForeignKey('Estanteria', 
                                on_delete=models.CASCADE,
                                related_name="estan")

    activo = models.BooleanField(default=True)
    creado = models.DateTimeField(auto_now_add=True)
    modificado = models.DateTimeField(auto_now=True)

    def delete(self, *args):
        self.activo = False
        self.save()
        return True

File: flutter/app_inventario/lib/models/detalle_producto.dart
class DetalleProductoModel {
  int? id;
  var producto;
  String? precio_costo; //Por el momento en String
  String? precio_venta; //Por el momento en String
  int? existenciasT;
  int? existenciasB;
  int? existencias;
  String? vencimiento;
  //var almacen;
  //var estanteria;

  DetalleProductoModel({
    this.id,
    this.producto,
    this.precio_costo,
    this.precio_venta,
    this.existenciasT,
    this.existenciasB,
    this.existencias,
    this.vencimiento,
  });

  factory DetalleProductoModel.fromJson(Map<String, dynamic> json) {
    return DetalleProductoModel(
      id: json['id'],
      producto: json['producto'],
      precio_costo: json['precio_costo'],
      precio_venta: json['precio_venta'],
      existenciasT: json['existenciasT'],
      existenciasB: json['existenciasB'],
      existencias: json['existencias'],
      vencimiento: json['vencimiento'],
      //almacen: json['almacen'],
      //estanteria: json['estanteria']
    );
  }

  int? get Id => id;
  String? get Precio_Costo => precio_costo;
  String? get Precio_Venta => precio_venta;
  int? get ExistenciasT => existenciasT;
  int? get ExistenciasB => existenciasB;
  int? get Existencias => existencias;
  String? get Vencimiento => vencimiento;

  dynamic toJson() => {
        'id': id,
        'producto': producto,
        'precio_costo': precio_costo,
        'precio_venta': precio_venta,
        'existenciasT': existenciasT,
        'existenciasB': existenciasB,
        'existencias': existencias,
        'vencimiento': vencimiento,
        //'almacen': almacen,
        //'estanteria': estanteria
      };
}
